inc/widgets/author-bio.php

```diff
@@ -25,28 +25,79 @@ function __construct() {
 	function widget( $args, $instance ) {
 
 		global $post;
-		$author_id = $post->post_author;
 
-		$output = '';
+		$authors = array();
+		$bios = '';
 
-		echo $args['before_widget'];
+		if ( is_singular() ) {
+			if ( function_exists( 'get_coauthors' ) ) {
+				$authors = get_coauthors( get_queried_object_id() );
+			} else {
+				$authors = array( get_user_by( 'id', get_queried_object()->post_author ) );
+			}
+		}
 
-		if ( ! empty( $instance['title'] ) ) {
-			echo $args['before_title'] . apply_filters( 'widget_title', $instance['title'] ) . $args['after_title'];
+		// make sure we have at least one bio before we show the widget
+		foreach ( $authors as $key => $author ) {
+			$bio = trim( $author->description );
+			if ( !is_author() && empty( $bio ) ) {
```

Comment:
`is_author()` is redundant here, because we're already checking `is_singular()` above